### PR TITLE
Simplify the structure of the Core and ClusterServer classes

### PR DESCRIPTION
**Pull request by lolski**
## What is the goal of this PR?

We're moving the initialisation of `channel` and `stub` object into the implementation class, `CoreClient` and `ClusterClient`. As a consequence the connection factory classes are no longer necessary and can be removed.

## What are the changes implemented in this PR?

1. Make `stub` overridable and can be instantiated with `TypeDBStub` or `ClusterServerStub` depending on the client type.
2. Remove connection factory classes.
3. Remove an unneeded `clusterServerStub` in `ClusterServerClient`.

### CODE REVIEW

File: connection/TypeDBClientImpl.java [hunk 1]
```diff
@@ -42,15 +42,11 @@
 
     private static final String TYPEDB_CLIENT_RPC_THREAD_NAME = "typedb-client-rpc";
 
-    private final ManagedChannel channel;
```
Comment (lolski) [inline connection/TypeDBClientImpl.java:45]: Move the definition of `channel` and `stub` into the respective concrete classes: `CoreClient` and `ClusterServer` client

File: connection/cluster/ClusterServerClient.java [hunk 2]
```diff
@@ -22,22 +22,65 @@
...
+        }
+    }
+
+    private ManagedChannel plainTextChannel(String address) {
+        return NettyChannelBuilder.forTarget(address)
+                .usePlaintext()
+                .build();
+    }
+
+    private ManagedChannel TLSChannel(String address, TypeDBCredential credential) {
```
Comment (alexjpwalker) [inline connection/cluster/ClusterServerClient.java:73]:
```suggestion
    private ManagedChannel tlsChannel(String address, TypeDBCredential credential) {
```

**Review by alexjpwalker — APPROVED** [review]

File: connection/cluster/ClusterServerClient.java (hunk 2 above)
Comment (lolski) [reply connection/cluster/ClusterServerClient.java]: Fixed in https://github.com/vaticle/typedb-client-java/pull/348/commits/c1da046274c41297d9449beb3a2a77a54e969370

File: connection/TypeDBConnectionFactory.java [hunk 3]
```diff
@@ -1,40 +0,0 @@
-/*
```
Comment (lolski) [inline connection/TypeDBConnectionFactory.java:1]: This class as well as their concrete implementations (for `CoreClient` and `ClusterServerClient`) can now be removed.

File: connection/cluster/ClusterServerClient.java [hunk 4]
```diff
@@ -22,22 +22,65 @@
...
+import io.grpc.ManagedChannel;
+import io.grpc.netty.GrpcSslContexts;
+import io.grpc.netty.NettyChannelBuilder;
+import io.netty.handler.ssl.SslContext;
+
+import javax.net.ssl.SSLException;
 
 class ClusterServerClient extends TypeDBClientImpl {
 
-    private final ClusterServerStub clusterServerStub;
```
Comment (lolski) [inline connection/cluster/ClusterServerClient.java:29]: This stub is a code smell that can be removed. We already have a stub in `TypeDBClientImpl` which we can use.

**Comment by lolski** [comment]
Closing this PR since the refactor is now included along with the [authentication token PR](https://github.com/vaticle/typedb-client-java/pull/350).

**Pull request closed without merging**